src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -118,6 +119,7 @@ public class MqttClient implements Closeable {
     private final List<AwsIotMqttClient> connections = new CopyOnWriteArrayList<>();
     private final Map<SubscribeRequest, AwsIotMqttClient> subscriptions = new ConcurrentHashMap<>();
     private final Map<MqttTopic, AwsIotMqttClient> subscriptionTopics = new ConcurrentHashMap<>();
+    private final Set<Integer> activeClientIds = new HashSet<>();
```

Comment:
What would we lose if we just tracked a counter an simply increased it, not worrying about if a connection that had a smaller number in the client id got closed?

Reply:
lose? nothing really. Just having large client IDs potentially.

Reply:
Not a strong opinion but I don't think we can have any guarantees on how large the client ids can get anyway, so I'd personally prefer to simplify this and not maintain a set.

Reply:
In this case, I prefer keeping the numbers smaller, especially for IoT Policies which may require customers to specify the exact client IDs that will be used. This method would let them specify a finite number of client IDs, say 1-10.